cabal-install/src/Distribution/Client/Config.hs

```diff
@@ -899,6 +899,10 @@ defaultHackageRemoteRepoKeys =
     "0a5c7ea47cd1b15f01f5f51a33adda7e655bc0f0b0615baa8e271f4c3351e21d"
   , -- Norman Ramsey (ZI8di3a9Un0s2RBrt5GwVRvfOXVuywADfXGPZfkiDb0=)
     "51f0161b906011b52c6613376b1ae937670da69322113a246a09f807c62f6921"
+  , -- Mathieu Boespflug (ydN1nGGQ79K1Q0nN+ul+Ln8MxikTB95w0YdGd3v3kmg=)
+    "be75553f3c7ba1dbe298da81f1d1b05c9d39dd8ed2616c9bddf1525ca8c03e48"
```

Comment:
How does `ydN1nGGQ79K1Q0nN+ul+Ln8MxikTB95w0YdGd3v3kmg=` become `be75553f3c7ba1dbe298da81f1d1b05c9d39dd8ed2616c9bddf1525ca8c03e48`?

Reply:
By converting encoding from Base64  to Hex???

Reply:
If I put `ydN1nGGQ79K1Q0nN+ul+Ln8MxikTB95w0YdGd3v3kmg=` [here](https://cryptii.com/pipes/base64-to-hex), I get `c9d3759c6190efd2b54349cdfae97e2e7f0cc6291307de70d18746777bf79268`.

I am probably doing a silly mistake.

In general, I wonder why the manual conversion, it seems error prone and makes reviewing the PR a bit more awkward, like in this case.

Reply:
As per https://github.com/haskell/hackage-security/tree/master#background-information the keyids  pasted are _hashes_ of the public keys. The root.json contains the mappings: https://github.com/haskell-infra/hackage-root-keys/pull/14/files

It is rather silly that the hash is longer than the key itself -- I think this schema is a holdover from when the keys used a different scheme (rsa) where they were much longer. Since we use ed25519 keys, the result is rather silly.

Reply:
This is indeed a bit silly. Nonetheless, putting it here makes the system easier to use as a whole.

Hackage as it exists is configured by assigning roles to key IDS (the hashes). The keys as they appear in the authoritative repo, however, don't include the key IDs. There's a command-line tool (`hackage-repo-tool`) that can calculate the key ID from a key file, but it's annoying to do that each time. Including the actual key value in the comment makes it easier to connect the public keys in the repository with the key IDs in this file.

It could definitely make sense to organize things differently, but I don't think that omitting this comment would help. And adding this documentation doesn't make it harder to refactor things later to make it unnecessary if one wants.

Reply:
When the switch to NIST PQC algorithms happens (probably several years from now), keys will again be _much_ larger that their hashes - and this silliness will go away.